src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,189 @@
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+import json
+import os
+import sas
+from  util import winnforum_testcase, makePpaAndPalRecordsConsistent
+import sas_testcase
+import hashlib
+from datetime import datetime, timedelta
+
+class FullActivityDumpMessageTestcase(sas_testcase.SasTestCase):
+
+    def setUp(self):
+        self._sas, self._sas_admin = sas.GetTestingSas()
+        self._sas_admin.Reset()
+
+    def tearDown(self):
+        pass
+    
+    def assertCbsdRecord(self, registration_request, grant_request, grant_response, recordData):
+        for index, device in enumerate(registration_request):
+            record_id = 'cbsd/'+ device['fccId']+'/'+ hashlib.sha1(device['cbsdSerialNumber']).hexdigest()
+            cbsd_record = [record['registration'] for record in recordData if record['id'] == record_id]
+            self.assertEqual(1, len(cbsd_record))
+            self.assertEqual(device['cbsdCategory'], cbsd_record[0]['cbsdCategory'])
+            self.assertEqual(device['fccId'], cbsd_record[0]['fccId'])
+            self.assertEqual(device['airInterface']['radioTechnology'],\
+                             cbsd_record[0]['airInterface']['radioTechnology'])
+            self.assertEqual(device['installationParam']['latitude'],\
+                             cbsd_record[0]['installationParam']['latitude'])
+            self.assertEqual(device['installationParam']['longitude'],\
+                             cbsd_record[0]['installationParam']['longitude'])
+            self.assertEqual(device['installationParam']['height'],\
+                             cbsd_record[0]['installationParam']['height'])
+            self.assertEqual(device['installationParam']['heightType'], \
+                             cbsd_record[0]['installationParam']['heightType'])
+            self.assertEqual(device['installationParam']['indoorDeployment'], \
+                             cbsd_record[0]['installationParam']['indoorDeployment'])
+            self.assertEqual(device['installationParam']['antennaGain'], \
+                             cbsd_record[0]['installationParam']['antennaGain'])     
+            self.assertEqual(device['installationParam']['antennaAzimuth'], \
+                             cbsd_record[0]['installationParam']['antennaAzimuth'])        
+            self.assertEqual(device['installationParam']['antennaDowntilt'], \
+                             cbsd_record[0]['installationParam']['antennaDowntilt'])       
+            self.assertEqual(device['installationParam']['antennaBeamwidth'], \
+                             cbsd_record[0]['installationParam']['antennaBeamwidth'])    
+                         
```

Comment:
The latest test spec says that we need to verify that "no banned content [n.12, R2-ISC-06, R2-ISC-07] is included." If you want to omit this for now, please add a comment like `TODO: Verify that no banned content [n.12,R2-ISC-06, R2-ISC-07] is included.` and make the update when possible.

Reply:
thank you very much for all your very useful comments, I updated the code to be configurable taking into account your comments.
though for this comment it is not clear what the information which can exist in the dump but it is a banned content according to [n.12,R2-ISC-06, R2-ISC-07], the serial number?